### PR TITLE
Audio integration to front end

### PR DESCRIPTION
**Pull request by nickyfantasy**
* Add flask end point
* Read and decode audio data

### CODE REVIEW

File: visualdl/server/lib.py [hunk 1]
```diff
@@ -155,6 +156,86 @@ def get_invididual_image(storage, mode, tag, step_index, max_size=80):
...
+
+    return result
+
+
+def get_audio_tag_steps(storage, mode, tag):
+    # remove suffix '/x'
+    res = re.search(r".*/([0-9]+$)", tag)
+    sample_index = 0
+    origin_tag = tag
+    if res:
```
Comment (daming-lu) [inline visualdl/server/lib.py:186]: What if the regex search failed? Should we add an 'else' to deal with it?

File: visualdl/server/lib.py [hunk 2]
```diff
@@ -155,6 +156,86 @@ def get_invididual_image(storage, mode, tag, step_index, max_size=80):
...
+            'query': query,
+        })
+
+    return res
+
+
+def get_individual_audio(storage, mode, tag, step_index, max_size=80):
+
+    with storage.mode(mode) as reader:
+        res = re.search(r".*/([0-9]+$)", tag)
```
Comment (daming-lu) [inline visualdl/server/lib.py:215]: Maybe we can abstract this " removing '/x' " logic into a method

File: visualdl/server/lib.py [hunk 3]
```diff
@@ -155,6 +156,86 @@ def get_invididual_image(storage, mode, tag, step_index, max_size=80):
...
+        if res:
+            offset = int(res.groups()[0])
+            tag = tag[:tag.rfind('/')]
+
+        audio = reader.audio(tag)
+        record = audio.record(step_index, offset)
+
+        data = np.array(record.data(), dtype='uint8')
+
+        tempfile = NamedTemporaryFile(mode='w+b', suffix='.wav')
```
Comment (daming-lu) [inline visualdl/server/lib.py:227]: Good to know this method from tempfile 😄

File: visualdl/server/visualDL [hunk 4]
```diff
@@ -245,6 +253,35 @@ def embeddings():
...
+                     tag)
+    result = gen_result(0, "", data)
+
+    return Response(json.dumps(result), mimetype='application/json')
+
+
+@app.route('/data/plugin/audio/individualAudio')
+def individual_audio():
+    mode = request.args.get('run')
+    print mode
```
Comment (jetfuel) [inline visualdl/server/visualDL:272]: Remove test print

**Review by jetfuel — APPROVED** [review]
LGTM

File: visualdl/server/lib.py (hunk 1 above)
Comment (nickyfantasy) [reply visualdl/server/lib.py:186]: in that case we just use default sample_index = 0

File: visualdl/server/lib.py (hunk 2 above)
Comment (nickyfantasy) [reply visualdl/server/lib.py:215]: sure

**Pull request merged**